pkg/sm/sm.go

```diff
@@ -98,40 +99,38 @@ func New(ctx context.Context, cancel context.CancelFunc, env env.Environment) *S
 		panic(fmt.Sprintf("error validating configuration: %s", err))
 	}
 
+	// Setup the default http client
 	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: cfg.API.SkipSSLValidation}
+	http.DefaultClient.Transport = http.DefaultTransport
+	http.DefaultClient.Timeout = cfg.Server.RequestTimeout
 
-	// setup logging
+	// Setup logging
 	ctx = log.Configure(ctx, cfg.Log)
 
-	// goroutines that log must be started after the log has been configured
 	util.HandleInterrupts(ctx, cancel)
 
-	// setup smStorage
+	// Setup storage
 	log.C(ctx).Info("Setting up Service Manager storage...")
-
-	waitGroup := &sync.WaitGroup{}
-
 	smStorage := &postgres.Storage{
 		ConnectFunc: func(driver string, url string) (*sql.DB, error) {
 			return sql.Open(driver, url)
 		},
 	}
 
+	// Decorate the storage with credentials encryption/decryption
+	encryptingDecorator := storage.EncryptingDecorator(ctx, &security.AESEncrypter{}, smStorage)
```

Comment:
it no longer matters if this is before or after storage initialization (it does not depend on storage.Open)